app/lib/notifications/firebase_messaging_callback_configurator.dart

```diff
@@ -126,6 +127,16 @@ class FirebaseMessagingCallbackConfigurator {
 }
 
 Future<void> _requestIOSPermission(BuildContext context) async {
+  const isIntegrationTest = bool.fromEnvironment('IS_INTEGRATION_TEST');
+  final isAndroid = PlatformCheck.isAndroid;
+  if (isIntegrationTest && isAndroid) {
+    // We skip requesting the permission in the integration test for Android
+    // because when running the tests with GitHub Actions Android emulators, the
+    // following exception will be thrown: "MissingPluginException(No
+    // implementation found for method Messaging#requestPermission on channel".
+    return;
+  }
```

Comment:
Here the build logs: https://github.com/SharezoneApp/sharezone-app/runs/6868516775?check_suite_focus=true

Reply:
We already have the permission because there is no need to request the permission on Android devices. This is also the reason why the method is called `_requestIOSPermission()`. But we need to rename or adjust this method for Android 13. Android 13 will introduce a similar feature as iOS has: you need to request the permission for sending push notification. 

https://techcrunch.com/2022/03/17/google-launches-android-13-developer-preview-2-psuh-notifications-opt-in/

I think the best solution would be to check if we need to call the `requestPermission` method on this device (based on the OS version). This would also remove the current solution with `isIntegrationTest`. But I would do this later when we adjust our app to Android 13.

Reply:
Wait but why do you even call isIntegrationTest in the method at all? And why don't we just do `if(platform== iOS)` _requestIOSPermissions? Or replace `kIsDriverTest` further above in the file with `isIntegrationTest`?

Reply:
I would suggest doing an `if(PlatformCheck.isIOS)` in the `_requestIOSPermissions()`